src/components/CalendarButton/CalendarButton.js

```diff
@@ -0,0 +1,116 @@
+import React, { Component } from 'react';
+import { findDOMNode } from 'react-dom';
+import PropTypes from 'prop-types';
+import styled, { css } from 'react-emotion';
+
+import { START_DATE } from 'react-dates/constants';
+
+import { RangePickerController } from '../Calendar';
+import Button from '../Button';
+
+class CalendarButton extends Component {
+  state = { startDate: null, endDate: null, focusedInput: null };
+  buttonRef = null;
+
+  _onDatesChange = ({ startDate, endDate }) => {
+    this.setState({ startDate, endDate });
+
+    if (startDate && endDate) {
+      this.props.onDatesRangeChange({ startDate, endDate });
+    }
+  };
+
+  _onFocusChange = focusedInput => this.setState({ focusedInput });
```

Comment:
Since setState doesn't return anything, I'd wrap it in `{}`.

Reply:
Done.